modules/shared/src/test/scala/org/tessellation/merkletree/MerkleTreeValidatorSuite.scala

```diff
@@ -0,0 +1,138 @@
+package org.tessellation.merkletree
+
+import cats.MonadThrow
+import cats.data.{NonEmptyList, NonEmptySet, Validated}
+import cats.effect.std.Console
+import cats.effect.{IO, Resource}
+import cats.syntax.functor._
+
+import scala.collection.immutable.{SortedMap, SortedSet}
+
+import org.tessellation.currency.schema.currency.{CurrencyIncrementalSnapshot, CurrencySnapshotInfo}
+import org.tessellation.kryo.KryoSerializer
+import org.tessellation.schema._
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.schema.epoch.EpochProgress
+import org.tessellation.schema.height.{Height, SubHeight}
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.security.Hashed
+import org.tessellation.security.hash.{Hash, ProofsHash}
+import org.tessellation.security.hex.Hex
+import org.tessellation.security.signature.Signed
+import org.tessellation.security.signature.signature.{Signature, SignatureProof}
+import org.tessellation.shared.sharedKryoRegistrar
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric.NonNegLong
+import weaver.MutableIOSuite
+
+object MerkleTreeValidatorSuite extends MutableIOSuite {
+
+  type Res = KryoSerializer[IO]
+
+  override def sharedResource: Resource[IO, Res] =
+    KryoSerializer.forAsync[IO](sharedKryoRegistrar)
+
+  private val hash = Hash("abc")
+
+  test("should succeed when use the same snapshot info for incremental global snapshot") { implicit ks =>
```

Comment:
imo - the test name should specify what it tests:
`should succeed when state matches snapshot's state proof`
it checks whether the state is valid, so it matches the proof
also, it fails when it cannot proof it, or it doesn't match the proof

Reply:
Changed it.